jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -650,45 +708,48 @@ def mos_niriss_parser(app, data_dir, table_viewer_reference_name='table-viewer')
-    try:
-        cat_fields = ['label', 'sky_centroid.ra', 'sky_centroid.dec']
-        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
-                                              delimiter=" ")
-    except KeyError:
-        # Older pipeline builds use different colname to distinguish sources
-        cat_fields[0] = 'id'
-        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
-                                              delimiter=" ")
-    for row in parsed_cat_fields:
-        cat_id_dict[int(row[0])] = (row[1], row[2])
+    # parse relevant information from source catalog if it exists
+    cat_id_dict = None
+    if files_by_labels[cat_key] != '':
+        cat_id_dict = {}
+        cat_file = files_by_labels[cat_key]
+        try:
+            cat_fields = ['label', 'sky_centroid.ra', 'sky_centroid.dec']
+            parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                                  delimiter=" ")
+        except KeyError:
+            # Older pipeline builds use different colname to distinguish sources
+            cat_fields[0] = 'id'
+            parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                                  delimiter=" ")
+        for row in parsed_cat_fields:
+            cat_id_dict[int(row[0])] = (row[1], row[2])
 
     # set up a dictionary of datasets to add to glue
     add_to_glue = {}
 
     # Parse images
     image_dict = {}
 
-    print("Loading: Images")
-    for image_file in files_by_labels["Direct Image"]:
-        # save label for table viewer
-        im_split = image_file.stem.split("_")[0]
-        pupil = fits.getheader(image_file, ext=0).get('PUPIL')
+    if "Direct Image" in files_by_labels:
+        print("Loading: Images")
+        for image_file in files_by_labels["Direct Image"]:
+            # save label for table viewer
+            im_split = image_file.stem.split("_")[0]
+            pupil = fits.getheader(image_file, ext=0).get('PUPIL')
 
-        image_label = f"Image {im_split} {pupil}"
-        image_dict[pupil] = image_label
+            image_label = f"Image {im_split} {pupil}"
+            image_dict[pupil] = image_label
 
-        # save information from data file
-        with fits.open(image_file) as temp:
-            data_iter = get_image_data_iterator(app, temp, "Image", ext=None)
-            data_obj = [d[0] for d in data_iter]  # We do not use the generated labels
-            image_data = data_obj[0]  # Grab the first one.
-            # TODO: Error if multiple found?
+            # save information from data file
+            with fits.open(image_file) as temp:
+                data_iter = get_image_data_iterator(app, temp, "Image", ext=None)
+                data_obj = [d[0] for d in data_iter]  # We do not use the generated labels
+                image_data = data_obj[0]  # Grab the first one.
+                # TODO: Error if multiple found?
```

Comment:
Is this TODO for this PR or future work?

Reply:
Great question - I think I should do it here (or alternatively at least think about it and decide it's not needed). I can do that after meetings today.